lib/maths/CTimeSeriesTestForSeasonality.cc

```diff
@@ -359,6 +364,11 @@ CSeasonalDecomposition CTimeSeriesTestForSeasonality::decompose() const {
     // is based on a number of criterion which are geared towards our modelling
     // techniques and are described in select.
 
+    if (this->checkInvariants() == false) {
+        LOG_ERROR(<< "Failed invariants. Not testing for seasonality.");
```

Comment:
Is it worth printing the 3 sizes involved here?  Or maybe `checkInvariants` could log them instead.

Reply:
I think this is a good suggestion. I think it is cleanest to add the logging to `checkInvariants` see [c3bd243](https://github.com/elastic/ml-cpp/pull/1611/commits/c3bd24322f36d3183512759eb54d497eed9a9e95).